File: src/main/java/com/github/wakingrufus/mastodon/client/OkHttpLogger.java
package com.github.wakingrufus.mastodon.client;

import lombok.extern.slf4j.Slf4j;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

import java.io.IOException;

@Slf4j
public class OkHttpLogger implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();

        long t1 = System.nanoTime();
        log.debug(String.format("Sending request %s on %s%n%s",
                request.url(), chain.connection(), request.headers()));

        Response response = chain.proceed(request);

        long t2 = System.nanoTime();
        log.trace(String.format("Received " + response.code() + " response for %s in %.1fms%n%s",
                response.request().url(), (t2 - t1) / 1e6d, response.headers()));

        return response;
    }
}

File: src/main/kotlin/com/github/wakingrufus/mastodon/client/OkHttpLogger.kt
package com.github.wakingrufus.mastodon.client

import mu.KLogging
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class OkHttpLogger : Interceptor {
    companion object : KLogging()

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()

        val t1 = System.nanoTime()
        logger.debug(String.format("Sending request %s on %s%n%s",
                request.url(), chain.connection(), request.headers()))

        val response = chain.proceed(request)

        val t2 = System.nanoTime()
        logger.trace(String.format("Received " + response.code() + " response for %s in %.1fms%n%s",
                response.request().url(), (t2 - t1) / 1e6, response.headers()))

        return response
    }
}